config/routes.rb

```diff
@@ -123,7 +123,12 @@
     resource :search,    only: :show
     resource :dashboard, only: :show, constraints: { format: /html|atom/ }
     resources :profiles, only: :show
-    resource :profile, only: [:edit, :update]
+    resource :profile, only: [:edit, :update] do
+      member do
+        get :delete
+        delete :destroy, as: :destroy
```

Comment:
Something to consider: This could be an entirely separate resource and concept in the app. "UserDeletion" perhaps, then it could be handled via a `new` and `create` action and not glommed onto the profile pages.